### PR TITLE
Add docs/README.md

### PR DESCRIPTION
**Pull request by fingolfin**
Then we can point people at it...

### CODE REVIEW

**Comment by fingolfin** [comment]
@wdecker could you try and see if these steps allow you to build the documentation, even with the `sorting = :nyt` in place?

**Comment by thofma** [comment]
Does this actually work? Does step 3) add the packages to the "standard" environment that one would get after restarting?

**Comment by fingolfin** [comment]
I did some limited testing, and it seemed to work but I admittedly was not very thorough. So the text may need to improved / extended.

**Comment by thofma** [comment]
Does not work for me in a clean environment.

**Comment by fieker** [comment]
All of this is confusing to the non-guru. E.g. after the `Pgk.activate`, doing `] st` will ONLY show the Doc-packages, I am not even sure that they will use the dev versions! Eg:
   [f1435218] Oscar v0.5.3-DEV `~/.julia/dev/Oscar`
becomes, after the activation: 
   [f1435218] Oscar v0.5.2
So I am not even running the doc against the "correct" Oscar version! On the other hand, I've already done "using Oscar", so I possibly still get the correct Oscar?
I think this approach needs some more work. Relying on the doc to be always in a clean environment is not practical with the current precompile times I'd say...

**Comment by wdecker** [comment]
This is frustrating:  I get tons of error messages starting like this:
```
julia> Oscar.build_doc()
[ Info: Precompiling Documenter [e30172f5-a6a5-5a46-863b-614d45cd2de4]
[ Info: Precompiling DocumenterMarkdown [997ab1e6-3595-5248-9280-8efb232c3433]
[ Info: Precompiling DocumenterCitations [daee34ce-89f3-4625-b898-19384cb65244]
[ Info: SetupBuildDirectory: setting up build directory.
[ Info: Doctest: running doctests.
┌ Error: doctest failure in ~/.julia/packages/AbstractAlgebra/Boo1X/docs/src/polynomial.md:175-229
```
There are too many to be printed here. Three examples:
```
Evaluated output:
│ 
│ (-30*x^2 - 43*x - 9)*y^3 + (-7*x^2 - 23*x - 7)*y^2 + (4*x^2 - 10*x - 3)*y + x^2 - 2*x
│ 
│ Expected output:
│ 
│ (-30*x^2 - 43*x - 9)*y^3 + (-7*x^2 - 23*x - 7)*y^2 + (4*x^2 - 10*x - 3)*y + (x^2 - 2*x)
```
```
┌ Warning: no docs found for 'DirectSum(::AbstractAlgebra.FPModule{T}...) where T <: RingElement' in `@docs` block in ~/.julia/packages/AbstractAlgebra/Boo1X/docs/src/direct_sum.md:21-24
│ ```@docs
│ DirectSum(::Vector{<:AbstractAlgebra.FPModule{T}}) where T <: RingElement
│ DirectSum(::AbstractAlgebra.FPModule{T}...) where T <: RingElement
│ ```
```

```
ERROR: LoadError: LoadError: Citation not found in bibliography: GLS10
Stacktrace:
```

**Comment by fieker** [comment]
I have a feeling we need to have a version of build_doc that can influence the strict flag. However, I am not quite sure how to achieve this. The make_local does a lot of initialisation, which does not need to be repeated over and over again

**Comment by fingolfin** [comment]
I can do that later today/tonight, and will also improve the docs and/or add more scripts to help with it

**Comment by fingolfin** [comment]
We should also explain how to setup a "good" dec environment, eg in DEVELOPMENT.md

**Comment by fieker** [comment]
see also #502

**Comment by fingolfin** [comment]
@thofma wrote:
> Does not work for me in a clean environment.

Could you please clarify what exactly did not work? I tried with with a fresh empty project and Julia 1.5 on my mac, and it worked fine. I am guessing you perhaps had something different in mind with "clean environment"?

**Comment by fingolfin** [comment]
@fieker wrote:

> All of this is confusing to the non-guru. 

It is? That's... just a few commands to enter? Not sure what's confusing about it? Perhaps you mean it's annoying to remember? 

Note, I am not saying this is *ideal*, and as I said, we could (and should) of course add some helpers to do it. But even if in the end `Oscar.build_doc()` does it all automatically somehow, I still think we absolutely need to document it and how to use it, in a way that we can point new (and old!) developers at it, otherwise they won't be able to benefit from these commands at all, and they'll be even worse of than with the "confusing" (?) instructions in this PR (namely with no instructions at all, and no hint at all how to build docs, nor how to build them "efficiently").

> E.g. after the `Pgk.activate`, doing `] st` will ONLY show the Doc-packages, I am not even sure that they will use the dev versions! Eg:
> [f1435218] Oscar v0.5.3-DEV `~/.julia/dev/Oscar`
> becomes, after the activation:
> [f1435218] Oscar v0.5.2

Hmmm, not sure I follow. That's not what I am seeing, so I am guessing we are doing something different at some point. Figuring out what might be illuminating in itself...

So I did this (removed some banners etc. for readability)
```
$ julia-1.5 --project=/tmp/proj-1.5
(proj-1.5) pkg> dev .
[ Info: Resolving package identifier `.` as a directory at `~/Projekte/OSCAR/Oscar.jl`.
Path `.` exists and looks like the correct package. Using existing path.
  Resolving package versions...
Updating `/private/tmp/proj-1.5/Project.toml`
  [f1435218] + Oscar v0.5.3-DEV `../../../Users/mhorn/Projekte/OSCAR/Oscar.jl`
Updating `/private/tmp/proj-1.5/Manifest.toml`
...

(proj-1.5) pkg> status
Status `/private/tmp/proj-1.5/Project.toml`
  [f1435218] Oscar v0.5.3-DEV `../../../Users/mhorn/Projekte/OSCAR/Oscar.jl`

(proj-1.5) pkg> add Revise
...

julia> using Revise, Oscar
...

julia> using Pkg ; Pkg.activate(joinpath(pkgdir(Oscar),"docs")) ; Pkg.instantiate()
 Activating environment at `~/Projekte/OSCAR/Oscar.jl/docs/Project.toml`

(docs) pkg> status
Status `~/Projekte/OSCAR/Oscar.jl/docs/Project.toml`
  [c3fe647b] AbstractAlgebra v0.18.0 `../../AbstractAlgebra.jl`
  [e30172f5] Documenter v0.27.2
  [daee34ce] DocumenterCitations v0.2.8
  [997ab1e6] DocumenterMarkdown v0.2.2
  [3e1990a7] Hecke v0.10.10 `../../Hecke.jl`
  [b3565e16] Kroki v0.1.0
  [2edaba10] Nemo v0.24.0 `../../Nemo.jl`
  [f1435218] Oscar v0.5.3-DEV `..`

julia> Oscar.build_doc()
[ Info: Precompiling DocumenterCitations [daee34ce-89f3-4625-b898-19384cb65244]
[ Info: SetupBuildDirectory: setting up build directory.
....

```


> So I am not even running the doc against the "correct" Oscar version! On the other hand, I've already done "using Oscar", so I possibly still get the correct Oscar?
> I think this approach needs some more work. Relying on the doc to be always in a clean environment is not practical with the current precompile times I'd say...

I guess I am less aware of this because I always have multiple Julia+Oscar instances running anyway (all with Revise) -- one to test the code etc. (running Julia 1.6) and one *exclusively* for regenerating the documentation (running Julia 1.5)

**Comment by thofma** [comment]
I think I misunderstood. Which parts are supposed to be run again after restarting? Only the last step? Or all steps except the first?

**Comment by thofma** [comment]
Regarding Claus comment with the `]status` not showing everything, this is a known problem, see https://github.com/JuliaLang/Pkg.jl/issues/961. Environments are stacked, but only packages of the top environment are displayed.

**Comment by fieker** [comment]
On Mon, Jun 28, 2021 at 11:49:52PM -0700, thofma wrote:
> Regarding Claus comment with the `]status` not showing everything, this is a known problem, see https://github.com/JuliaLang/Pkg.jl/issues/961. Environments are stacked, but only packages of the environment are displayed.
> 
Can I pop the stack?

**Comment by thofma** [comment]
After skimming over https://docs.julialang.org/en/v1/manual/code-loading/#Environments-1, I don't think so.

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        using Pkg ; Pkg.add("Revise")
+
+2. Start a fresh Julia session and load Revise before Oscar:
+
+        using Revise, Oscar
+
+3. Ensure all packages required for building the documentation are available:
+
+        using Pkg ; Pkg.activate(joinpath(pkgdir(Oscar),"docs")) ; Pkg.instantiate()
+
```
Comment (thofma) [inline docs/README.md]:
```suggestion
```
This is not necessary anymore.

**Pull request merged**